src/index.ts

```diff
@@ -13,18 +14,27 @@ export function createMiddlewareAddon(
     notebookApi: IVSCodeNotebook,
     getClient: () => LanguageClient | undefined,
     traceInfo: (...args: any[]) => void,
-    cellSelector: string,
+    cellSelector: string | DocumentSelector,
     notebookFileRegex: RegExp,
     pythonPath?: string,
     trace?: (message: string) => void
 ): MiddlewareAddon {
-    return new NotebookMiddlewareAddon(
-        notebookApi,
-        getClient,
-        traceInfo,
-        cellSelector,
-        notebookFileRegex,
-        pythonPath,
-        trace
-    );
+    // We have two types of middleware addons.
+    // - Python based addon that python extension creates. It consumes
+    // all events for python selector
+    // - Notebook based addon taht jupyter extension creates. It processes
+    // events for a jupyter (python) notebook
+    if (cellSelector === 'python') {
```

Comment:
Per the type change above does this also need to see if it's part of an array or a DocumentFilter with .language === python?

Reply:
It will eventually. For now it works with the test (and the python extension).

My next iteration is going to modify the tests to use per notebook selectors. I'll change the python test to use the selectors for python.